### PR TITLE
Add commandline support (issue #5)

### PR DESCRIPTION
**Pull request by skjolber**
Download commandline executable and run a scan.

### CODE REVIEW

**Comment by skjolber** [comment]
Kanskje litt uheldig at man alltid må være på siste versjon av dependency-check? Kan legge til valgfri parameter for å sette versjonen.

File: README.md [hunk 1]
```diff
@@ -105,6 +110,34 @@ workflows:
...
+Configure a job
+
+```yaml
+workflows:
+  version: 2.1
+  build:
+    jobs:
+      - owasp/commandline_owasp_dependency_check:
+          executor: java_11
+          context: global
```
Comment (tommybo-entur) [inline README.md]: No need for context in example

File: README.md [hunk 2]
```diff
@@ -105,6 +110,34 @@ workflows:
...
+The default OWASP arguments is `--scan ./`, for using other commands, add an `arguments` parameter as so:
+
+```yaml
+workflows:
+  version: 2.1
+  build:
+    jobs:
+      - owasp/commandline_owasp_dependency_check:
+          executor: java_11
+          context: global
```
Comment (tommybo-entur) [inline README.md]: No need for context in example

File: src/@orb.yml [hunk 3]
```diff
@@ -40,6 +40,23 @@ examples:
...
+
+      orbs:
+        owasp: entur/owasp@0.0.4
+
+      workflows:
+        owasp:
+          jobs:
+            - owasp/commandline_owasp_dependency_check:
+                executor: java_11
+                context: global
```
Comment (tommybo-entur) [inline src/@orb.yml]: No need for context in example

File: src/@orb.yml [hunk 4]
```diff
@@ -40,6 +40,23 @@ examples:
                 executor: java_11
                 context: global
 
+  commandline_project:
```
Comment (tommybo-entur) [inline src/@orb.yml]: It's maybe sufficient calling the example command_line?

File: src/@orb.yml [hunk 5]
```diff
@@ -153,22 +170,84 @@ commands:
...
                 root: .
                 paths:
                   - Reports/OWASP
+  with_commandline:
+    description: |
+      Run a set of steps with the OWASP Dependency-Check command-line tool. Run from ~/.owasp/dependency-check/bin/dependency-check.sh.
+    parameters:
+      steps:
+        type: steps
+      version_url:
```
Comment (tommybo-entur) [inline src/@orb.yml]: Why should this be customizable?

File: src/@orb.yml [hunk 6]
```diff
@@ -153,22 +170,84 @@ commands:
...
+        description: Location of latest Dependency-Check command-line tool version (i.e. 5.2.1)
+        type: string
+        default: "https://jeremylong.github.io/DependencyCheck/current.txt"
+      executable_url:
```
Comment (tommybo-entur) [inline src/@orb.yml]: As an alternative you could expect the executing image to contain dependenct_check.sh. Perhaps even have a separate command for installing owasp.

**Review by tommybo-entur — APPROVED** [review]
Looks good. Added some thoughts for you to consider.

File: src/@orb.yml (hunk 5 above)
Comment (skjolber) [reply src/@orb.yml]: Storing the URLs using `environment` in the job would be an alternative - better?

File: src/@orb.yml (hunk 4 above)
Comment (skjolber) [reply src/@orb.yml]: Renaming maven and gradle examples correspondingly

File: src/@orb.yml (hunk 5 above)
Comment (tommybo-entur) [reply src/@orb.yml]: Sure

**Review by Pavloro — APPROVED** [review]

File: src/@orb.yml (hunk 6 above)
Comment (skjolber) [reply src/@orb.yml]: I'll save that for a later version + might be sensible to also look at customizing the command-line arguments also for maven and gradle builds.

**Pull request merged**